File: 프로그래머스/3/17676. ［1차］ 추석 트래픽/［1차］ 추석 트래픽.py
def t2ms(t):
    s = t.split()
    h,m,sec = s[1].split(":")
    second = int(sec.split(".")[0])*1000
    ms = int(sec.split(".")[1])
    end = int(h)*60*60*1000+int(m)*60*1000+second+ms
    dur = float(s[2][:-1])*1000
    start = end - int(dur) + 1
    return (start,end)
def solution(lines):
    ans = 0
    arr = [t2ms(l) for l in lines]
    ts = []
    for s,e in arr:
        ts.append(s)
        ts.append(e)
    for t in ts:
        te = t + 1000
        cnt = sum(1 for s,e in arr if e >= t and s < te)
        ans = max(cnt,ans)
    return ans

File: 프로그래머스/3/17676. ［1차］ 추석 트래픽/［1차］ 추석 트래픽.java
import java.util.*;
class Solution {
    class Pair {
        int a;
        int b;
        public Pair(int a, int b) {
            this.a = a;
            this.b = b;
        }
    }
    
    public int solution(String[] lines) {
        int ans = 0;
        Pair[] arr = new Pair[lines.length];
        for (int i = 0; i < lines.length; i++) {
            arr[i] = s2ms(lines[i]);
        }
        List<Integer> timestamp = new ArrayList<>();
        for (Pair p : arr) {
            int a = p.a;
            int b = p.b;
            timestamp.add(a);
            timestamp.add(b);
        }
        for (int t : timestamp) {
            int cnt = 0;
            for (Pair p : arr) {
                int start = p.a;
                int end = p.b;
                if (t <= end && t + 1000 > start) cnt++;
            }
            ans = Math.max(ans, cnt);
        }
        return ans;
    }
    
    public Pair s2ms(String str) {
        String[] split = str.split(" ");
        String[] t = split[1].split(":");
        int h = Integer.valueOf(t[0])*60*60*1000;
        int m = Integer.valueOf(t[1])*60*1000;
        String[] s = t[2].split("\\.");
        int sec = Integer.valueOf(s[0])*1000;
        int ms = Integer.valueOf(s[1]);
        float duration = Float.valueOf(split[2].substring(0,split[2].length()-1))*1000;
        int end = h+m+sec+ms;
        int start = end - (int) duration + 1;
        return new Pair(start,end);
    }
}